pkg/config/validation/validation.go

```diff
@@ -2896,3 +2896,46 @@ func validateLocalities(localities []string) error {
 
 	return nil
 }
+
+// ValidateMeshNetworks validates meshnetworks.
+func ValidateMeshNetworks(meshnetworks *meshconfig.MeshNetworks) (errs error) {
+	for name, network := range meshnetworks.Networks {
+		if err := validateNetwork(network); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, fmt.Sprintf("invalid network %v:", name)))
+		}
+	}
+	return
+}
+
+func validateNetwork(network *meshconfig.Network) (errs error) {
+	for _, n := range network.Endpoints {
+		switch e := n.Ne.(type) {
+		case *meshconfig.Network_NetworkEndpoints_FromCidr:
+			if err := ValidateIPSubnet(e.FromCidr); err != nil {
+				errs = multierror.Append(errs, err)
+			}
+		case *meshconfig.Network_NetworkEndpoints_FromRegistry:
+			r := e.FromRegistry
+			if strings.EqualFold(r, "kubernetes") && r != "Kubernetes" {
```

Comment:
Is `Kubernetes` a special hard coded name or as long as it matches the secret even `kubernetes` is fine?

Reply:
there are two possible values: matching the secret (multicluster) or built-in registries Kubernetes and MCP. Unfortunately the API is just a string so we don't have a way to distinguish a typo

Reply:
The primary cluster's ID is `Kubernetes` by default.  But it can be changed

Reply:
Can we find out if the default has changed? Otherwise this is a breaking change.
If we want to make it a breaking change, then we can be more forceful.

Reply:
Yes, we can get it from `Server.clusterID`. So i think it need pass in